.github/workflows/on-pr-debug.yml

```diff
@@ -56,33 +35,61 @@ jobs:
       matrix:
         compiler: [gcc, clang]
         sanitize: [address, thread]
-        pooled_delivery: [ON, OFF]
+        pooled_dispatch: [pool, NO-pool]
     uses: ./.github/workflows/build-test.yml
     with:
       server_version: main
-      type: Debug
+      type: RelWithDebInfo
       compiler: ${{ matrix.compiler }}
       sanitize: ${{ matrix.sanitize }}
-      lib_msg_delivery: ${{ matrix.pooled_delivery }}
+      pool_dispatch: ${{ matrix.pooled_dispatch }}
 
-  sanitize-write-deadline:
-    name: "Sanitize address (write_deadline)"
+  coverage-TLS:
+    name: "Coverage: TLS"
+    strategy:
+      fail-fast: false
+      matrix:
+          pooled_dispatch: [pool, NO-pool]
+          write_deadline: [write_deadline, NO-write_deadline]
     uses: ./.github/workflows/build-test.yml
     with:
+      coverage: ON
       type: RelWithDebInfo
-      sanitize: address
       server_version: main
-      lib_write_deadline: ON
+      compiler: gcc
+      tls: TLS
+      verify_host: verify_host
+      pool_dispatch: ${{ matrix.pooled_dispatch }}
+      write_deadline: ${{ matrix.write_deadline }}
+    secrets:
+      CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
 
-  sanitize-no-host-verify:
-    name: "Sanitize address (NO force_host_verification)"
+  coverage-NO-verify_host:
```

Comment:
Should we have a single coverage run (basically with the defaults of the library that has TLS/enfore_hostname_verification/etc..)? 
Or is it that we would get a much lower score (for all code that is in the "#else" of those "#ifdef") without these new runs?

Reply:
Right. What prompted this was the refactoring of deliveries - the thread pool refactor code generated a lot of missing coverage comments since it was not ran with coverage.

My thinking: the coverage overhead is relatively small (relative to sanitize) so it's ok to run a bigger matrix. For the same reason, the coverage runs are probably sufficient to assert functional correctness (no need to dial down the parameters, for example). We still run the no-coverage tests for the default build config.

FWIW, I failed to package them into a single compact matrix statement, but the current job set is probably more intention revealing anyway then the matrix would have been.